lib/shelley/src/Cardano/Wallet/Shelley/Network/Blockfrost.hs

```diff
@@ -521,6 +526,14 @@ withNetworkLayer tr network np project k = do
                         }
             void $ LN.lightSync (MsgLightLayerLog >$< tr) lightSyncSource follower
 
+    syncProgress :: BF.ClientConfig -> SlotNo -> IO SyncProgress
+    syncProgress bfConfig s = runBFM bfConfig do
+        BF.Block {_blockSlot} <- BF.getLatestBlock
+        let latestSlot = maybe 0 BF.unSlot _blockSlot
+            currentSlot = fromIntegral (unSlotNo s)
+            percentage = (currentSlot * 100) % latestSlot
+        pure $ Syncing $ Quantity $ clipToPercentage percentage
```

Comment:
Fine for an MVP implementation.

For the full implementation, we have to take into account the `SyncTolerance`. It's a command line parameter recorded in [data ServeArgs](https://github.com/input-output-hk/cardano-wallet/blob/4a62cef2c6b055e297c31fb93dd07af646049e6c/lib/shelley/exe/cardano-wallet.hs#L189). See [Cardano.Wallet.Primitive.SyncProgress](https://github.com/input-output-hk/cardano-wallet/blob/bd8a530e46e3ffa74cd293f128d231abb7073aca/lib/core/src/Cardano/Wallet/Primitive/SyncProgress.hs#L10) for semantics. For Blockfrost, we don't have to ask for the current wall-clock time, but we may need a `TimeInterpreter` to convert the slot difference into a time difference that can be compared to `SyncTolerance`.

Reply:
Ok, added to the [tech-debt issue](https://input-output.atlassian.net/browse/ADP-1651)